common/lib/client/auth.js

```diff
@@ -97,7 +108,33 @@ var Auth = (function() {
 			Logger.logAction(Logger.LOG_ERROR, 'Auth()', msg);
 			throw new Error(msg);
 		}
-	}
+	};
+
+	/**
+	 * Update the authentication options for this client;
+	 * this will also update these options on the realtime connection
+	 * so that they take effect immediately
+	 * @param options
+	 */
+	Auth.prototype.setOptions = function(options) {
```

Comment:
`authorise()` just forces the generation of a new token using current options.

`setOptions` sets the options on which `authorise()` acts. What if you are setting a token explicitly in `options` ? What if you change the token capabilities by specifying different params in the `authURL` ? You can't change either of those things just by calling `authorise()`.